### ISSUE TITLE
[Bug]: referralSetReferees  missing volume

### ISSUE THREAD

**Issue by bwallacee** [vegaprotocol/vega#10462]
### Problem encountered

I am testing teams / referrals, and see then we are returning `totalRefereeNotionalTakerVolume: "0"`  from `referralSetReferees` for the referral page, But I see in Fills that i have a Taker notional of 10350000. 

### Observed behaviour

```
node:{id: "157a3e53f85cada7ebd83fa06e2cbcaf44c8b1d6f4582abbd05c838cc56d485f",…}
aggressor:"SIDE_BUY"
buyOrder: "99e038b8cdd8dd7ef27d361aedd4b33b72a536c22157606b20aa78f83ad10dc1"
buyer: {id: "17a219a00dc683708bf4eb79840cf4d44bfc4f864ee11fa970e19d4e59dfa174", __typename: "Party"}
id: "17a219a00dc683708bf4eb79840cf4d44bfc4f864ee11fa970e19d4e59dfa174"
__typename: "Party"
buyerFee: {makerFee: "1035000", infrastructureFee: "5175", liquidityFee: "0", makerFeeReferralDiscount: "0",…}
createdAt: "2024-01-21T08:19:29.247653Z"
id: "157a3e53f85cada7ebd83fa06e2cbcaf44c8b1d6f4582abbd05c838cc56d485f"
market: {id: "4716db2e177f9119e0378ed3793483fd0de520610e57b9481ea777b923e84e86", __typename: "Market"}
id: "4716db2e177f9119e0378ed3793483fd0de520610e57b9481ea777b923e84e86"
__typename: "Market"
price: "10350000"
```
This is using market-sim against vega version `v0.74.0-preview.2`

![image](https://github.com/vegaprotocol/vega/assets/133853298/fbfce886-9c73-4997-94e4-42668772470b)
![image](https://github.com/vegaprotocol/vega/assets/133853298/84e2d525-85a7-4f67-99f8-2ff6790d7252)

### Expected behaviour

I would have expected `totalRefereeNotionalTakerVolume` to return the same volume as seen in the Fills table.

### Steps to reproduce

```bash
1. Refer a key
2. with referred key, get a taker fill
3. with referee key view your referees
4. Observe error
```

### Software version

v0.74.0-preview.2

**Comment by bwallacee** [comment]
[genesis.txt](https://github.com/vegaprotocol/vega/files/14010360/genesis.txt)
[replaytxt.txt](https://github.com/vegaprotocol/vega/files/14010362/replaytxt.txt)

**Comment by guoguojin** [comment]
Having checked and discussed with @bwallacee, we have come to the conclusion that this is not a bug in the API.

Ben created referral set as a team, and thought the stats would be produced, but this API requires a non-team referral set and a referral program for stats to be generated I believe.

I have pointed out the correct teams API to use for team member stats.

Our unit tests for the data retrieval for this API is incomplete though and didn't test that part of the query correctly. I will create a separate PR containing the updates to those tests.

Closing this issue for now.
